packages/logger/tests/unit/injectLambdaContext.test.ts

```diff
@@ -239,4 +240,333 @@ describe('Inject Lambda Context', () => {
+      await handler(testEvent, context);
+
+      // Assess
+      expect(console.info).toHaveBeenCalledTimes(1);
+      expect(console.info).toHaveLoggedNth(
+        1,
+        expect.objectContaining({
+          message: 'Hello, world!',
+          correlation_id: '12345-test-id',
+          ...getContextLogEntries(),
+        })
+      );
+      expect(logger.getCorrelationId()).toBe('12345-test-id');
+    });
+
+    it('should set correlation ID manually', () => {
+      // Prepare
+      const logger = new Logger();
+
+      // Act
+      logger.addContext(context);
+      logger.setCorrelationId('12345-test-id');
+      logger.info('Hello, world!');
+
+      // Assess
+      expect(console.info).toHaveBeenCalledTimes(1);
+      expect(console.info).toHaveLoggedNth(
+        1,
+        expect.objectContaining({
+          message: 'Hello, world!',
+          correlation_id: '12345-test-id',
+          ...getContextLogEntries(),
+        })
+      );
+
+      expect(logger.getCorrelationId()).toBe('12345-test-id');
+    });
+
+    it('should warn when correlationIdPath is provided but no search function is available', async () => {
+      // Prepare
+      const logger = new Logger(); // No search function provided
+      const warnSpy = vi.spyOn(logger, 'warn');
+
+      // Act - Use middleware which will internally call setCorrelationIdFromPath
+      const handler = middy(async () => {
+        logger.info('Hello, world!');
+      }).use(
+        injectLambdaContext(logger, {
+          correlationIdPath: 'headers.x-correlation-id',
+        })
+      );
+
+      await handler(testEvent, context);
+
+      // Assess
+      expect(warnSpy).toHaveBeenCalledWith(
+        'correlationIdPath is set but no search function was provided. The correlation ID will not be added to the log attributes.'
+      );
+    });
+
+    it('should not set correlation ID when search function returns falsy value', async () => {
+      // Prepare
+      const logger = new Logger({ correlationIdSearchFn: search });
+
+      // Act - Use middleware which will internally call setCorrelationIdFromPath
+      const handler = middy(async () => {
+        logger.info('Hello, world!');
+      }).use(
+        injectLambdaContext(logger, {
+          correlationIdPath: 'headers."non-existent"',
+        })
+      );
+
+      await handler(testEvent, context);
+
+      // Assess
+      expect(logger.getCorrelationId()).toBeUndefined();
+      expect(console.info).toHaveBeenCalledTimes(1);
+      expect(console.info).toHaveLoggedNth(
+        1,
+        expect.not.objectContaining({
+          correlation_id: expect.anything(),
+        })
+      );
+    });
+
+    it('should propagate correlation ID to child loggers', () => {
+      // Prepare
+      const logger = new Logger();
+      logger.setCorrelationId('parent-correlation-id');
+
+      // Act
+      const childLogger = logger.createChild();
+      childLogger.info('Hello from child!');
+
+      // Assess
```

Comment:
Correlation ID is stored in temporary attributes, so if there's a test that assesses they're propagated this will propagated too.

A more useful test for this feature would be to check that the search function is propagated to children instead.

Reply:
we can do both